src/docfx/build/xref/XrefMap.cs

```diff
@@ -15,34 +13,24 @@ internal class XrefMap
         private readonly IReadOnlyDictionary<string, Lazy<ExternalXrefSpec>> _externalXrefMap;
         private readonly IReadOnlyDictionary<string, InternalXrefSpec> _internalXrefMap;
 
-        private static ThreadLocal<Stack<(string uid, string propertyName, Document parent)>> t_recursionDetector = new ThreadLocal<Stack<(string, string, Document)>>(() => new Stack<(string, string, Document)>());
-
         public XrefMap(Context context, Docset docset, RestoreFileMap restoreFileMap)
         {
             _internalXrefMap = InternalXrefMapBuilder.Build(context);
             _externalXrefMap = ExternalXrefMapLoader.Load(docset, restoreFileMap);
         }
 
-        public (Error error, string href, string display, IXrefSpec xrefSpec) Resolve(string uid, SourceInfo<string> href, string displayPropertyName, string text, Document relativeTo)
+        public (Error error, string href, IXrefSpec xrefSpec) Resolve(string uid, SourceInfo<string> href)
```

Comment:
can JsonSchemaTransformer.cs line 236 be updated to use this method instead because it does not need a link?

Reply:
Not really, because `XrefMap.Resolve` takes `uid` as input, which `JsonSchemaTransformer` only has `href`. And also `DependencyResolver` seems to be a better entry point for resolving.

Reply:
the content there MUST be an `uid`, not an `href`. The `xref` content type only cares about retrieving an `XrefSpec`, not building up an URL or display name

Reply:
ok, if content can only be `uid` there

Reply:
hmm... We also handled the dependency map within `DependencyResolver.ResolveAbsoluteXref`, so it is a better entry point.